### PR TITLE
added support for forever command. added new database layer to make s…

### PR DESCRIPTION
**Pull request by hthayko**
…ingle db connection that is shared among all methods

### CODE REVIEW

File: src/lib/database.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+var Sequelize = require('sequelize');
+var sql;
+var sqlInited;
+
+module.exports = {
```
Comment (stevenzeiler) [inline src/lib/database.js:5]: Here we can simply export the `sql` connection object without a `initSql` function.

File: src/prior.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 'use strict';
-var Sequelize = require('sequelize');
+var DB = require('./lib/database');
```
Comment (stevenzeiler) [inline src/prior.js:2]: Could change to `var sql = require('./lib/database')`

File: index.js [hunk 3]
```diff
@@ -54,6 +54,13 @@ commander
   });
 
 commander
+  .command('forever <ipp> <dbUrl>')
+  .description('run crawl forever starting from ipp(-s flag will be turned on automatically')
```
Comment (wilsonianb) [inline index.js]: missing close parenthesis

File: src/lib/database.js (hunk 1 above)
Comment (hthayko) [inline src/lib/database.js:5]: that would mean no use of initSql arguments and instead using environment variables for dbUrl and log. Will do that on next release.

**Comment by hthayko** [comment]
:+1:

File: index.js (hunk 3 above)
Comment (stevenzeiler) [inline index.js]: It all looks good to me once the parenthesis typo is fixed.

**Comment by hthayko** [comment]
:+1:

File: src/lib/error.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
+	/**
+	 * @param  {[type]}
+	 * @return {Boolean}
+	 * returns a predicate, that checkes if given error is of the given type. 
+	 * Predicate can be used for passing to bluebird .catch(errorPredicate, errorHandler)
+	 */
+	isOfType : function(type){
```
Comment (wilsonianb) [inline src/lib/error.js]: Is this used anywhere?

Comment (hthayko) [inline src/lib/error.js]: it was used. Then we decided not to do anything special for database errors so this is redundant as of now. Should I remove it ?

Comment (wilsonianb) [inline src/lib/error.js]: If we don't plan on using it, I'd say remove it.

**Comment by wilsonianb** [comment]
Think you could fold these extra commits?

**Comment by hthayko** [comment]
what do you mean ?

**Comment by wilsonianb** [comment]
LGTM :+1:

**Comment by hthayko** [comment]
:+1:

**Pull request merged**